lib/views/post/widgets/post_info_bar.dart

```diff
@@ -1,22 +1,31 @@
 import 'package:flutter/material.dart';
 import 'package:pr0gramm/api/dtos/item/item.dart';
 import 'package:pr0gramm/entities/enums/vote.dart';
-import 'package:pr0gramm/entities/post_info.dart';
 import 'package:pr0gramm/services/vote_service.dart';
 import 'package:pr0gramm/views/post/widgets/op_info.dart';
-import 'package:pr0gramm/views/post/widgets/post_buttons.dart';
-import 'package:pr0gramm/widgets/global_inherited.dart';
+import 'package:pr0gramm/views/post/widgets/post_vote.dart';
 
 class PostInfoBar extends StatelessWidget {
   final Item item;
+  final VoteService _voteService = VoteService.instance;
 
   PostInfoBar({Key key, this.item}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return Row(
       children: [
-        PostButtons(item: item, direction: Axis.horizontal,),
+        FutureBuilder<Vote>(
```

Comment:
Show the vote button in all cases. If the data is not yet available show as if the user didn't vote for the item yet and update the button once the data is available.

Reply:
resolved with beeing able to provide null as initialVote to the animation service by defaulting it to focusedColor. since initialVote will change if it is loaded the buttons + the service are reconstructed and the new data is now in view

Reply:
908061a